src/Jackalope/ObjectManager.php

```diff
@@ -818,12 +818,12 @@ public function save()
             $this->executeOperations($this->operationsLog);
 
             // loop through cached nodes and commit all dirty and set them to clean.
-            if (isset($this->objectsByPath['Node'])) {
-                foreach ($this->objectsByPath['Node'] as $node) {
+            foreach ($this->objectsByPath as $nodes) {
```

Comment:
lets add a loop for saving all version things if the transport is a GenericVersionInterface or what its called. that should loop over Version/Version and whatever other keys are created by the VersionHandler adding non-Node classes.